nomad/state/state_store_host_volumes_test.go

```diff
@@ -163,6 +163,12 @@ func TestStateStore_HostVolumes_CRUD(t *testing.T) {
 	must.NoError(t, err)
 	got = consumeIter(iter)
 	must.MapLen(t, 3, got, must.Sprint(`expected 3 volumes remain`))
+
+	prefix := vol.ID[:30] // sufficiently long prefix to avoid flakes
+	iter, err = store.HostVolumesByIDPrefix(nil, "*", prefix, SortDefault)
```

Comment:
only covering wildcard `"*"`, not a specific namespace?

Reply:
The specific namespace case was trivial because it's just calling an already-testing function, but can't hurt to add a case for it. Done.

Reply:
I tend to be paranoid about stuff like this, less about the code as currently written, but more to guard against potential future regressions.